controller/Review.php

```diff
@@ -151,16 +151,21 @@ public function getItem(): void
             ]);
 
             // make sure the responses data are compliant to QTI definition
-            $itemData['data']['responses'] = array_filter($responsesData, static function ($key) use ($responsesData) {
-                return array_key_exists('qtiClass', $responsesData[$key]) && array_key_exists('serial', $responsesData[$key]);
-            }, ARRAY_FILTER_USE_KEY);
+            $itemData['data']['responses'] = array_filter(
+                $responsesData,
+                static function (array $dataEntry): bool {
+                    return array_key_exists('qtiClass', $dataEntry)
+                        && array_key_exists('serial', $dataEntry)
+                        && $dataEntry['qtiClass'] !== 'modalFeedback';
+                }
+            );
         }
 
         $response['content'] = $itemData;
         $response['baseUrl'] = $itemPreviewer->getBaseUrl();
         $response['success'] = true;
 
-        $this->returnJson($response);
+        $this->setSuccessJsonResponse($response);
```

Comment:
This, and the above `setSuccessJsonResponse` call change the response interface.
`setSuccessJsonResponse` puts the data under `data` property, whilst the `returnJson` sets the `data` as-is.

Reply:
Reverted it back, I hope you don't mind, @Alroniks.